sdk/all/src/main/java/io/opentelemetry/sdk/resources/internal/EnvironmentResourceProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.resources.internal;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.EnvironmentResource;
+import io.opentelemetry.sdk.resources.Resource;
+
+/** {@link ResourceProvider} for automatically configuring {@link EnvironmentResource}. */
+public final class EnvironmentResourceProvider implements ResourceProvider {
+  @Override
+  public Resource createResource(ConfigProperties config) {
+    return EnvironmentResource.create(config);
+  }
+
+  @Override
+  public int order() {
+    // A high order ensures that the environment resource takes precedent over the resources from
+    // other ResourceProviders
+    return 10;
```

Comment:
Consider making this a larger number:
```suggestion
    return 10_000;
```
or even `1_000_000`. We have a couple of resource providers in instrumentation/contrib that use values 100/200/1000, and they should definitely run before this one. Also a large order number doesn't really change anything, and it does provide some extra flexibility when it comes to customization.